src/Data/Foreign/Generic/Enum.purs

```diff
@@ -0,0 +1,126 @@
+module Data.Foreign.Generic.EnumEncoding where
+
+import Prelude
+
+import Control.Alt ((<|>))
+import Data.Foreign (F, Foreign, ForeignError(..), fail, readString, toForeign)
+import Data.Foreign.Generic.Types (Options, SumEncoding(..))
+import Data.Generic.Rep (class Generic, Argument, Constructor(Constructor), NoArguments(NoArguments), Product, Rec, Sum(Inr, Inl), from, to)
+import Data.Symbol (class IsSymbol, SProxy(..), reflectSymbol)
+import Partial.Unsafe (unsafeCrashWith)
+
+-- | A generic function to be used with "Enums", or sum types with only no-argument constructors. This is used for decoding from strings to one of the constructors, combined with the `constructorTagTransform` property of `SumEncoding`.
+genericDecodeEnum
+  :: forall a rep
+   . Generic a rep
+  => GenericDecodeEnum rep
+  => Options
+  -> Foreign
+  -> F a
+genericDecodeEnum opts = map to <<< decodeEnum opts
+
+-- | A generic function to be used with "Enums", or sum types with only no-argument constructors. This is used for encoding to strings from one of the constructors, combined with the `constructorTagTransform` property of `SumEncoding`.
+-- |
+-- | For example:
+-- |
+-- | ```purescript
+-- | data Fruit = Apple | Banana | Frikandel
+-- | derive instance geFruit :: Generic Fruit _
+-- | instance eFruit :: Encode Fruit where
+-- |   encode = genericEncodeEnum defaultOptions
+genericEncodeEnum
+  :: forall a rep
+   . Generic a rep
+  => GenericEncodeEnum rep
+  => Options
+  -> a
+  -> Foreign
+genericEncodeEnum opts = encodeEnum opts <<< from
+
+-- | A type class for type representations that can be used for decoding to an Enum. Only the sum and no-argument constructor instances are valid, while others provide a `Fail` constraint to fail in compilation.
+-- |
+-- | For example:
+-- |
+-- | ```purescript
+-- | data Fruit = Apple | Banana | Frikandel
+-- | derive instance geFruit :: Generic Fruit _
+-- | instance dFruit :: Decode Fruit where
+-- |   decode = genericDecodeEnum defaultOptions
+-- | ```
+class GenericDecodeEnum a where
+  decodeEnum :: Options -> Foreign -> F a
+
+-- | A type class for type representations that can be used for encoding from an Enum. Only the sum and no-argument constructor instances are valid, while others provide a `Fail` constraint to fail in compilation.
+class GenericEncodeEnum a where
+  encodeEnum :: Options -> a -> Foreign
+
+instance sumGenericDecodeEnum
+  :: (GenericDecodeEnum a, GenericDecodeEnum b)
+  => GenericDecodeEnum (Sum a b) where
+  decodeEnum opts f = Inl <$> decodeEnum opts f <|> Inr <$> decodeEnum opts f
+
+instance ctorNoArgsGenericDecodeEnum
+  :: IsSymbol name
+  => GenericDecodeEnum (Constructor name NoArguments) where
+  decodeEnum {sumEncoding: TaggedObject {constructorTagTransform}} f = do
+    tag <- readString f
+    unless (constructorTagTransform tag == ctorName) $
```

Comment:
ah, yeah. the ctorname is already transformed, but i guess transforming the tag itself is kinda wonky...